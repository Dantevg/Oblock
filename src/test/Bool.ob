var Bool
Bool = {
	"true" = .clone()
	"false" = .clone()
	
	"!" () => if this == Bool.true: Bool.false else Bool.true
	toString() => if this == Bool.true: "Bool:true" else "Bool:false"
}

print(Bool.true.toString())
print(Bool.false.toString())
print((!Bool.true).toString())
print(!Bool.true == Bool.false)
