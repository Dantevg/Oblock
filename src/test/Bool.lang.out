Bool:true
Bool:false
Bool:false
true
